### ISSUE TITLE
server is http

### ISSUE THREAD

**Issue by alnavv** [ets-labs/python-dependency-injector#304]
Hello,

all your docs are stored in a HTTP server, however, many company proxies/AntiVir block any HTTP site.

Just mentioning that moving to a https  will make your docs available to a bigger audience (including myself :)).

Regards,
   A.

**Comment by rmk135** [comment]
Hi @alnavv , good point, thanks. I consider how to move the docs to use secure connection.

**Comment by rmk135** [comment]
Done.

Readthedocs, where the docs is hosted, provides SSL certificates for custom domains in a partnership with Cloudflare. I enabled force usage of ``https://`` and updated links in docs, readmes and examples to lead to ``https://`` explicitly.

The fixes are published in version `4.0.5`.

Appreciate you've brought this up. Thanks for the contribution.

I'm closing the issue. Please, feel free to re-open or comment if needed.

**Comment by alnavv** [comment]
Hi, 

I've just quickly checked it. And unfortunately, i am still getting this common error:

_**Secure Connection Failed

An error occurred during a connection to python-dependency-injector.ets-labs.org. SSL received a record that exceeded the maximum permissible length.

Error code: SSL_ERROR_RX_RECORD_TOO_LONG**_

**Comment by alnavv** [comment]
A few more comments:
This place is accessible:
**https://readthedocs.org/projects/python-dependency-injector/**
however, any other https link inserted there is not (same error than above).

Besides, your main page is blocked: 
**https://python-dependency-injector.ets-labs.org/**
 with the same error:

_An error occurred during a connection to python-dependency-injector.ets-labs.org. SSL received a record that exceeded the maximum permissible length.

Error code: SSL_ERROR_RX_RECORD_TOO_LONG_

hope it helps!

**Comment by rmk135** [comment]
@alnavv , got it, thanks. I'm unfamiliar with this error. Will investigate today.

**Comment by rmk135** [comment]
@alnavv , can you please check what ssl version are you using? Is it TLS 1.2/1.3?

**Comment by rmk135** [comment]
I've tried to find something about mentioned error. There could be different reasons. I've also scanned the docs with SSL Lab test and found something on readthedocs side.

I have opened an issue in readthedocs project - https://github.com/readthedocs/readthedocs.org/issues/7585. Let's see if guys there could help to fight this problem.

**Comment by alnavv** [comment]
The TSL depends on the site (which negotiates). I have an extension in firefox to show the version in use. For instance, on this (github) page is 1.3. However, in all the other problematic pages it shows a "?" meaning that it can not retrieve it.
I would say is definitely an issue on your side (i mean, your provider). 
Let me know if you need anything else on my side. Happy to help.

**Comment by rmk135** [comment]
@alnavv, I've deployed docs on different platform and did a bunch of other optimizations. Can you, please, try https://d2zgs0mzwm2xxk.cloudfront.net/?

If that works I'll consider moving the docs there.

PS: the firefox extension you're talking about, is it https://addons.mozilla.org/en-US/firefox/addon/https-everywhere/?

**Comment by javiertoledos** [comment]
Not sure if you're aware of but I'm seeing that the ets-labs.org domain is parked at Godaddy, the certificate error means the original hosting is not longer serving the documentation and the browser is complaining because Godaddy is now serving the site (a parked domain).

**Comment by rmk135** [comment]
Hey @javiertoledos , the docs might be in storming state right now. I did some experiments with Cloudflare today and now rolling that back.

Original setup I had is that docs was served by readthedocs server, domain was parked on godaddy and there was a CNAME to readthedocs. Certificates were provided by Cloudflare as a part of their partnership with readthedocs.

**Comment by alnavv** [comment]
> @alnavv, I've deployed docs on different platform and did a bunch of other optimizations. Can you, please, try https://d2zgs0mzwm2xxk.cloudfront.net/?
> 
> If that works I'll consider moving the docs there.
> 
> PS: the firefox extension you're talking about, is it https://addons.mozilla.org/en-US/firefox/addon/https-everywhere/?

the extension is **IndicateTLS**

https://github.com/jannispinter/indicatetls

**Comment by alnavv** [comment]
> @alnavv, I've deployed docs on different platform and did a bunch of other optimizations. Can you, please, try https://d2zgs0mzwm2xxk.cloudfront.net/?
> 
> If that works I'll consider moving the docs there.
> 
> PS: the firefox extension you're talking about, is it https://addons.mozilla.org/en-US/firefox/addon/https-everywhere/?
and yes, it works in that link!

**Comment by rmk135** [comment]
@alnavv 
> the extension is IndicateTLS
Got it, thanks.

Did you close the issue accidentally?

**Comment by rmk135** [comment]
I'll re-open the issue

**Comment by rmk135** [comment]
@alnavv so while https://d2zgs0mzwm2xxk.cloudfront.net/ works for you I think that the issue with SSL_ERROR_RX_RECORD_TOO_LONG_ is related to SNI  support. At least this is the only difference I see between current deployment and https://d2zgs0mzwm2xxk.cloudfront.net/.

**Comment by rmk135** [comment]
@javiertoledos , can you check if things are working fine now?

**Comment by rmk135** [comment]
@alnavv , can I, please, ask you to test https://di.ets-labs.org/?

**Comment by javiertoledos** [comment]
> @javiertoledos , can you check if things are working fine now?

it´s working well again. Both https://python-dependency-injector.ets-labs.org/examples/index.html and  https://di.ets-labs.org/

**Comment by rmk135** [comment]
@javiertoledos great, thank you

**Comment by alnavv** [comment]
hello @rmk135, 

https://di.ets-labs.org/  -> does **NOT** work:
SSL_ERROR_RX_RECORD_TOO_LONG

https://python-dependency-injector.ets-labs.org/examples/index.html -> does **NOt** work:
same isue

https://d2zgs0mzwm2xxk.cloudfront.net/ -> it does work!

**Comment by alnavv** [comment]
switching to chromium (instead of firefox), the error is : 

This site can’t provide a secure connectiondi.ets-labs.org sent an invalid response.
ERR_SSL_PROTOCOL_ERROR

**Comment by javiertoledos** [comment]
@alnavv if you're getting those errors it seems to be something specific to your environment. ets-labs.org site is using TLS 1.2 and TLS 1.3 and it seems to be well configured. 

![image](https://user-images.githubusercontent.com/284327/97114691-99cb5d00-16b7-11eb-8751-b0c063c405cc.png)

If you're behind a proxy/av firewall/corporate network, I  think you should test in a different network to discard network issues. Maybe it's a stale dns issue on your end?

**Comment by rmk135** [comment]
@alnavv @javiertoledos thanks for checking. Here is my update:

I tried different ssl checking tools and most of them provide good ratings for https://di.ets-labs.org/. The funny thing is that https://di.ets-labs.org/ and https://d2zgs0mzwm2xxk.cloudfront.net/ both point to the same location. Moreover, they both show very similar results in ssl testing.

My best guess I have now is that I have ets-labs.org domain parked on godaddy while https://di.ets-labs.org/ is served by AWS Cloudfront. I will try to move the domain to AWS Route 53. Many people complain about different problems with godaddy, may be it will help.

**Comment by alnavv** [comment]
hi @rmk135, @javiertoledos 

It seems the underlying reason is McAffe/Cylan. In macOS, it works if you are not using the VPN. In Windows machines, even without the VPN, the page is not accesible. I am actually working in a debian VM in windows host (virtualbox). 

There is nothing on our side we can do. It happens  with very specific webpages, I mean is not a generic issue, for which i presume there should be something on the server side configuration that our antiVirs detect as a thread. 

hope it helps.

**Comment by rmk135** [comment]
@alnavv , got it, thanks for the explanation. I'm really wiling to get things fixed.

Can you, please, try https://di.ets-labs.org/ now?

I've changed the way DNS routing works. Domain now is parked at AWS Route 53, docs is served from AWS Cloudfront and certificate is issued by AWS Certificate Manager.

PS: Unfortunately, I have to work in a "blind mode" cause I can not reproduce the issue on my environment. Appreciate if you could help with testing. I believe we're quite close to get that resolved.

**Comment by javiertoledos** [comment]
@alnavv the issue could be some root CA not properly working and appliance failing to inspect ssl traffic, or your computers clock may be out of date, or it could be an IDS false positive that your AV is handling as a thread. I'm not fully aware at that level what's going on, but I would say the issue is local since you mentioned there are more sites that gives you that issue. I tested from my home and work and the site is displayed properly, so I believe it depends more on your infrastructure than the site's infrastructure right now. 

In the past, these kind of errors were usually caused by misconfigured intermediate certificate chain, but this is not the case (I checked again with another ssl checker tool and I also configured sites with cloudfront and ACM and they manage it well).

The day I mentioned it didn't work was because the domain was misconfigured, but that's not longer the case, I would try to figure out with a local network IT member if there's something else. On my side, I would discard something server side based on what I mentioned earlier.

**Comment by alnavv** [comment]
hi @rmk135,

unfortunately, your latest https://di.ets-labs.org/ does not work either. However, https://d2zgs0mzwm2xxk.cloudfront.net/ still works.

@javiertoledos, you might be completely right regarding the "local" issue. No disagreement here. However, my initial consideration is that our IT configuration (let's call it like that) block access to your webpage and that IT configuration is shared by more than 70000 employees, and therefore you might want to overcome the issue on your side, given the fact that nothing can be done on ours (that was my initial suggestion).  
This issue affects a few other webpages, but very few. In general, only the ones that rely on unsecured "HTTP".
.

**Comment by rmk135** [comment]
@alnavv , thanks for checking.

Can you, please, check https://di.ets-labs.org/ one more time? Please, drop browser cache and check that certificate is provided by LetsEncrypt (see screenshot, example with chrome).

<img width="1409" alt="Screenshot 2020-10-27 at 11 25 04" src="https://user-images.githubusercontent.com/1742049/97325214-1fcadd80-1849-11eb-880a-7660b91917ad.png">

> unfortunately, your latest https://di.ets-labs.org/ does not work either. However, https://d2zgs0mzwm2xxk.cloudfront.net/ still works.

These two are the same servers with different virtual names and certificates. I checked the differences between the certificates. It seems that working-one certificate has a different root. May be there is a lack of trust to AWS as a CA... Not sure.  I've tried to provision a certificate from a new provider, LetsEncrypt. It provides clean and straightforward chain (see screenshot 2 and 3).

![FireShot Capture 037 - SSL Certificate Checker - Diagnostic Tool - DigiCert com - www digicert com](https://user-images.githubusercontent.com/1742049/97324865-b8ad2900-1848-11eb-8f0f-73a4d260631b.png)
![FireShot Capture 040 - SSL Checker - www sslshopper com](https://user-images.githubusercontent.com/1742049/97324870-bb0f8300-1848-11eb-830b-b69b39190aa7.png)

**Comment by rmk135** [comment]
@alnavv ,  did you have a chance to try https://di.ets-labs.org/ with new certificates :point_up:?

**Comment by rmk135** [comment]
I have moved https://python-dependency-injector.ets-labs.org/ docs to AWS. 

Current distribution uses AWS S3, Cloudfront, Route 53 and CM with the certificate from LetsEncrypt. Such distribution shows better results in ssl and website performance analyzers. Moving docs from ReadTheDocs also provides a way to remove the ads.

As of https://di.ets-labs.org/, it was started to experiment with the distribution. I'm going to kill it soon.

I convey respect and appreciation to @alnavv @javiertoledos . Thank you for bringing up the issue and actively participate in the discussion. I don't know if the https problem has been resolved, but everything that is done already has very positive impact on the project.

I'm closing the issue for now. Thanks again.

Best,
Roman

**Comment by rmk135** [comment]
PS: @alnavv , if https://python-dependency-injector.ets-labs.org/ still doesn't work for you, use https://d2zgs0mzwm2xxk.cloudfront.net/. It doesn't go anywhere.

**Comment by alnavv** [comment]
Hi @rmk135,

unfortunately, only https://d2zgs0mzwm2xxk.cloudfront.net/ works for me.
All the other links show an "?" where the TLS version number should be displayed. At the same time no certificate (firefox):
(if you need any other screenshoot just let me know)  
![a](https://user-images.githubusercontent.com/22129598/97623169-8e649280-1a25-11eb-88e6-0a61b70e43b6.png)

**Comment by alnavv** [comment]
and here for the https://d2zgs0mzwm2xxk.cloudfront.net/ link. You can see the option of displaying the certificate atthe left side:
![b](https://user-images.githubusercontent.com/22129598/97624067-db953400-1a26-11eb-8c40-f80f07420c0d.png)

**Comment by rmk135** [comment]
Hey @alnavv. Got it. Sorry to hear that. Here is what I see in Firefox:

<img width="1138" alt="Screenshot 2020-10-29 at 15 37 43" src="https://user-images.githubusercontent.com/1742049/97624022-e044f280-19fc-11eb-82d5-b538e31b84bc.png">
<img width="1172" alt="Screenshot 2020-10-29 at 15 37 20" src="https://user-images.githubusercontent.com/1742049/97624029-e0dd8900-19fc-11eb-8de8-1829bfb1ae84.png">

PS: On your screenshot you use https://di.ets-labs.org/ - please, make sure you use https://python-dependency-injector.ets-labs.org/

**Comment by rmk135** [comment]
@alnavv , I guess it's sort of cache or something. On your screenshot you tried https://di.ets-labs.org/ - but it is already dead.

Please, try to clean the cache and visit  https://python-dependency-injector.ets-labs.org/

**Comment by alnavv** [comment]
no luck. I drop the caches several times, switch to windows, but all the same. Including this: https://python-dependency-injector.ets-labs.org/.
I will check with some colleagues, to make sure that this issue still affects them.

**Comment by rmk135** [comment]
@alnavv , got it, thank you. Drop me a line, please, if you find anything. I'm out of ideas what else to fix now. I've re-opened the issue.

**Comment by alnavv** [comment]
@rmk135,

in macos, without vpn, works. In windows does not work in both cases (with/without vpn).
In macos, without vpn, chrome gives a usefull message:

_"blocked by mcaffe", the  URL is listed in categories that are  not allowd ...
URL categories: parqued domain
reputation: unverified_

I am wondering if the title itself, "injector", has anything to do with that :-)

**Comment by rmk135** [comment]
@alnavv , that's useful. Thank you. I'll try to find more information about that. _URL categories: parqued domain_ - that might be what @javiertoledos was talking about. Domain ets-labs.org was parked at GoDaddy and when you visited it you got to GoDaddy ads page. If so that explains why it was categorized like that and the rule _the URL is listed in categories that are not allowd ..._ also makes sense.

**Comment by rmk135** [comment]
I have submitted a ticket to Mcafee to get the docs re-categorized to "Software/Hardware" category. Waiting for their resolution. Fingers crossed.

Ticket number is #2225903.

<img width="1224" alt="Screenshot 2020-10-30 at 21 47 52" src="https://user-images.githubusercontent.com/1742049/97768508-e79df600-1af9-11eb-9538-a935b2654b59.png">
<img width="1277" alt="Screenshot 2020-10-30 at 21 48 12" src="https://user-images.githubusercontent.com/1742049/97768514-ebca1380-1af9-11eb-94dc-25993190e691.png">

**Comment by rmk135** [comment]
@alnavv , hey, can you try now? Mcafee has re-categorized the docs url.

<img width="681" alt="Screenshot 2020-11-02 at 12 19 30" src="https://user-images.githubusercontent.com/1742049/97898378-cad60e00-1d05-11eb-9e83-df893a5dc681.png">

**Comment by rmk135** [comment]
@alnavv , hey, did you try https://python-dependency-injector.ets-labs.org/ recently?

**Comment by alnavv** [comment]
Hi @rmk135,

it finally works :-)

**Comment by rmk135** [comment]
Wonderful!

Thank you again for getting through all of this with me 😊

**Comment by alnavv** [comment]
@rmk135,

your welcome! But I think the content really deserved the effort. 
Keep up the good work!

**Comment by rmk135** [comment]
Yep, it was a useful journey. Thanks again @alnavv .
